### PR TITLE
Feature/data emission

### PR DESCRIPTION
**Pull request by maerzhase**
This PR adds support for:
- `$fx.emit()` 
- `$fx.context`

For now controllers are only disabled when update "code-driven". 

# Example projects

- For testing data emission: https://dev.fxhash-dev.xyz/generative/slug/dollarfx.emit-emission-one
- For testing code driven project: https://dev.fxhash-dev.xyz/generative/slug/code-driven-project

# Todo

- [ ] Implement dedicate minimal minting experience when all params "code-driven"
- [x] Make sure "context" is set everywhere properly (e.g. when open in new tab)

### CODE REVIEW

File: src/containers/MintWithTicket/MintWithTicketPage.tsx [hunk 1]
```diff
@@ -91,6 +91,7 @@ export function MintWithTicketPageRoot({ token, ticketId, mode }: Props) {
     },
     {
       autoRefresh: withAutoUpdate,
+      urlParams: new URLSearchParams("fxcontext=minting")
```
Comment (maerzhase) [inline src/containers/MintWithTicket/MintWithTicketPage.tsx]: We can pass in search params that are appended to the url of the runtime

File: src/components/Testing/ControlsTest.tsx [hunk 2]
```diff
@@ -40,28 +40,43 @@ export const ControlsTest = forwardRef<ControlsTestRef, ControlsTestProps>(
...
     return (
       <div className={classes.container}>
+        {allParamsCodeDriven && (
+          <p className={classes.codeDrivenNote}>
+            <i className="fa-solid fa-triangle-exclamation" aria-hidden />
+            <span>
+              All params of this artwork are defined as "code-driven". This will
+              enable a dedicated minting experience for collectors.
+            </span>
+          </p>
```
Comment (maerzhase) [inline src/components/Testing/ControlsTest.tsx:56]: When testing params and all params are code driven we notify the user that we detected the dedicated minting experience

File: src/hooks/useRuntimeController.ts [hunk 3]
```diff
@@ -355,14 +359,22 @@ export const useRuntimeController: TUseRuntimeController = (
...
+      },
+      options?.urlParams
+    )
   }, [project.cid, runtime.details.stateHash.hard])
 
+  useMessageListener("fxhash_emit:params:update", (e: any) => {
+    const { params } = e.data.data
+    updateParams(params)
+  })
+
```
Comment (maerzhase) [inline src/hooks/useRuntimeController.ts:377]: we update the runtime state when we receive params emission

**Review by ciphrd — APPROVED** [review]
LGTM

**Pull request merged**